LKExp-Firmware_update-Serial_update/BOOTLOADER/lib/QspiMemory.h

```diff
@@ -0,0 +1,52 @@
+/**
+******************************************************************************
+* @file    LekaFirmware.h
+* @author  Yann Locatelli
+* @brief   Firmware operational test code
+******************************************************************************
+*/
+
+/* Prevent recursive inclusion -----------------------------------------------*/
+#ifndef __LEKAFIRMWARE_H__
+#define __LEKAFIRMWARE_H__
+
+#include "QSPI.h"
+#include "mbed.h"
+#include "MX25L51245G_config.h"
+
+#define CMD_READ 0x03
+#define CMD_WRITE 0x02
+#define CMD_ERASE 0x20
+#define CMD_RDSR 0x5
+#define CMD_WREN 0x6
+#define CMD_RSTEN 0x66
+#define CMD_RST 0x99
+#define STATUS_REG_SIZE 2
+#define BIT_WIP 0x1
+#define BIT_WEL 0x2
```

Comment:
je préfère les `const` ou `constexpr qspi_inst_t CMD_READ`, ça permet au compilateur d'optimiser et de faire des check pendant les compilations.

Comme la liste des commandes est longue, on peut même imaginer un `enum class`:

```
enum class QspiInstruction: qspi_inst_t { // ou QspiCommand
    READ  = 0x03,
    WRITE = 0x02,
    ERASE = 0x20,
	// etc.
};
```

```suggestion
enum class QspiInstruction: qspi_inst_t { // ou QspiCommand
    READ  = 0x03,
    WRITE = 0x02,
    ERASE = 0x20,
	// etc.
};
#define STATUS_REG_SIZE 2
#define BIT_WIP 0x1
#define BIT_WEL 0x2
```
et pareil pour les autres qui ne sont pas des commandes mais qui doivent à minima être des `constexpr <T>`

Reply:
J'ai repris l'[exemple QSPI](https://os.mbed.com/docs/mbed-os/v6.2/apis/spi-apis.html) de mbed sur le sujet.

Après comme il s'agit de l'exploration et que le but est de "faire fonctionner" pour bien saisir les mécanismes et les paramètres à régler, j'ai pas pris le temps de mettre correctement en forme. Ca sera adapté dans la version officielle.